core-bundle/tests/Contao/Database/ResultTest.php

```diff
@@ -0,0 +1,124 @@
+            $this->assertSame('SELECT from test', $result->query);
+            $this->assertSame(0, $result->count());
+            $this->assertSame([], $result->fetchAllAssoc());
+            $this->assertFalse($result->fetchAssoc());
+            $this->assertSame([], $result->fetchEach('test'));
+            $this->assertFalse($result->fetchRow());
+        }
+    }
+
+    public function testMultipleRows()
+    {
+        $data = [
+            ['field' => 'value1'],
+            ['field' => 'value2'],
+        ];
+
+        $resultStatement = new Result(new DoctrineArrayStatement($data), 'SELECT from test');
+        $resultArray = new Result($data, 'SELECT from test');
+
+        foreach ([$resultStatement, $resultArray] as $result) {
+            $this->assertFalse($result->isModified);
+            $this->assertSame(1, $result->numFields);
+            $this->assertSame(2, $result->numRows);
+            $this->assertSame('SELECT from test', $result->query);
+            $this->assertSame(2, $result->count());
+            $this->assertSame($data, $result->fetchAllAssoc());
+            $this->assertSame($data[0], $result->reset()->fetchAssoc());
+            $this->assertSame($data[1], $result->fetchAssoc());
+            $this->assertSame(['value1', 'value2'], $result->fetchEach('field'));
+            $this->assertSame(array_values($data[0]), $result->reset()->fetchRow());
+            $this->assertSame(array_values($data[1]), $result->fetchRow());
+            $this->assertFalse($result->fetchRow());
+
+            $this->assertSame($data[0], $result->reset()->row());
+            $this->assertSame(array_values($data[1]), $result->last()->row(true));
+
+            $this->assertSame('value1', $result->first()->field);
+            $this->assertFalse($result->prev());
+            $this->assertSame('value2', $result->next()->field);
+            $this->assertSame('value1', $result->prev()->field);
+            $this->assertSame('value2', $result->last()->field);
+            $this->assertFalse($result->next());
+        }
+    }
+
+    public function testFetchRowAndAssoc()
+    {
+        $data = [
+            ['field' => 'value1'],
+            ['field' => 'value2'],
+        ];
+
+        $resultStatement = new Result(new DoctrineArrayStatement($data), 'SELECT from test');
+        $resultArray = new Result($data, 'SELECT from test');
+
+        foreach ([$resultStatement, $resultArray] as $result) {
+            $this->assertSame(['field' => 'value1'], $result->fetchAssoc());
+            $this->assertSame(['field' => 'value1'], $result->row());
+            $this->assertSame('value1', $result->field);
+            $this->assertNull($result->{'0'});
+
+            $this->assertSame(['value2'], $result->fetchRow());
+            $this->assertSame(['value2'], $result->row());
+            $this->assertSame('value2', $result->{'0'});
+            $this->assertNull($result->field);
```

Comment:
While writing the tests I noticed a strange behavior with accessing the data after calling `fetchRow()`, as you can see here the behavior of `row()` and `__get()` changes if the last row was fetched via `fetchRow()`.

I think we should probably change this to work the same as `fetchAssoc()` (apart from the different return value of `fetchRow()` itself).

Reply:
Agreed.

Reply:
Done in ac75d2630c5da1e96079c2e60cf33673c9ba5d01